lib/ClangImporter/ImportDecl.cpp

```diff
@@ -2036,11 +2041,76 @@ namespace {
         enumDecl->addMember(rawValueGetter);
         enumDecl->addMember(rawValue);
         enumDecl->addMember(rawValueBinding);
-        
         result = enumDecl;
+
+        // Set up the domain error member to be lazily added
+        if (enumInfo.isErrorEnum()) {
+          auto clangDomainIdent = enumInfo.getErrorDomain();
+          auto &clangSema = Impl.getClangSema();
+          clang::LookupResult lookupResult(
+              clangSema, clang::DeclarationName(clangDomainIdent),
+              clang::SourceLocation(),
+              clang::Sema::LookupNameKind::LookupOrdinaryName);
+
+          if (!clangSema.LookupName(lookupResult, clangSema.TUScope)) {
+            // Couldn't actually import it as an error enum, fall back to enum
+            break;
+          }
+
+          auto clangDecl = lookupResult.getAsSingle<clang::NamedDecl>();
+          if (!clangDecl) {
+            // Couldn't actually import it as an error enum, fall back to enum
+            break;
+          }
+          auto swiftDecl = Impl.importDecl(clangDecl);
+          if (!swiftDecl || !isa<ValueDecl>(swiftDecl)) {
+            // Couldn't actually import it as an error enum, fall back to enum
+            break;
+          }
+          SourceLoc noLoc = SourceLoc();
+          bool isStatic = true;
+          bool isImplicit = true;
+
+          DeclRefExpr *domainDeclRef = new (swiftCtx) DeclRefExpr(
+              ConcreteDeclRef(cast<ValueDecl>(swiftDecl)), {}, isImplicit);
+          auto stringTy = swiftCtx.getStringDecl()->getDeclaredType();
```

Comment:
NSError uses NSString and CFErrorRef uses CFStringRef, so I think it would be fine to support just those two.